tests/test_logutils.py

```diff
@@ -0,0 +1,141 @@
+import re
+import sys
+import unittest
+import warnings
+import numpy as np
+
+from six import StringIO
+from os import unlink
+from sklearn.model_selection import GridSearchCV
+from sklearn.ensemble import RandomForestClassifier
+from tempfile import NamedTemporaryFile
+
+from skll import (close_and_remove_logger_handlers, get_skll_logger,
+                  orig_showwarning)
+
+
+def trigger_sklearn_warnings():
+    """
+    Do something that will trigger ``sklearn`` warnings.
+
+    This should trigger a couple ``FutureWarning``s about
+    ``n_estimators``'s default value and specifying ``cv``.
+    """
+
+    (GridSearchCV(RandomForestClassifier(), {"max_depth": [None, 1, 2]})
+     .fit(np.array([1, 1, 1, 2, 2, 2]).reshape(6, 1), [2, 2, 2, 3, 3, 3]))
+
+
+class TestLogUtils(unittest.TestCase):
+
+    def setUp(self):
+        self.TEMP_FILES = []
+        self.TEMP_FILE_PATHS = []
+        self.LOGGERS = []
+
+    def tearDown(self):
+        self.reset()
+
+    def reset(self):
+        for i, temp_file in enumerate(self.TEMP_FILES):
+            temp_file.close()
+            del self.TEMP_FILES[i]
+        for i, temp_file_path in enumerate(self.TEMP_FILE_PATHS):
+            unlink(temp_file_path)
+            del self.TEMP_FILE_PATHS[i]
+        for i, logger in enumerate(self.LOGGERS):
+            close_and_remove_logger_handlers(logger)
+            del self.LOGGERS[i]
+        warnings.showwarning = orig_showwarning
+
+
+    def testGetSKLLLogger(self):
+        self.reset()
+
+        temp_file = NamedTemporaryFile("w", delete=False)
+        temp_file.close()
+        self.TEMP_FILES.append(temp_file)
+        self.TEMP_FILE_PATHS.append(temp_file.name)
+        logger = get_skll_logger("testGetSKLLLogger", temp_file.name)
+        self.LOGGERS.append(logger)
+
+        # Send a regular log message
+        msg1 = "message 1"
+        logger.info(msg1)
+
+        # Send a regular log message
+        msg2 = "message 2"
+        logger.info(msg2)
+
+        with open(temp_file.name) as temp_file:
+            log_lines = temp_file.readlines()
+            assert log_lines[0].endswith("INFO - {}\n".format(msg1))
+            assert log_lines[1].endswith("INFO - {}\n".format(msg2))
+
+
+    def testGetSKLLLoggerWithWarning(self):
```

Comment:
can we make the name of the methods all lowercase with underscores rather than camel case.

Reply:
Sure. I think that should be fine. I'm not sure why but it seems like the `unittest` module expects/uses camel case, though (e.g. `setUp`).